wagtail/core/query.py

```diff
@@ -173,11 +174,10 @@ def not_page(self, other):
         return self.exclude(self.page_q(other))
 
     def type_q(self, klass):
-        content_types = ContentType.objects.get_for_models(*[
+        content_types = get_content_types_for_models(*[
```

Comment:
I think this would be clearer if we if we just passed `for_concrete_model=True` into `get_for_model` as we already know it's a page